dotnet/src/SemanticKernel.Skills/Skills.OpenAPI/Authentication/BasicAuthenticationProvider.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.SemanticKernel.Skills.OpenAPI.Authentication;
+
+/// <summary>
+/// Retrieves authentication content (e.g. username/password, API key) via the provided delegate and
+/// applies it to HTTP requests using the "basic" authentication scheme.
+/// </summary>
+public class BasicAuthenticationProvider
+{
+    private readonly Func<Task<string>> _content;
+
+    /// <summary>
+    /// Creates an instance of the <see cref="BasicAuthenticationProvider"/> class.
+    /// </summary>
+    /// <param name="content">Delegate for retrieving the authentication content.</param>
+    public BasicAuthenticationProvider(Func<Task<string>> content)
```

Comment:
In the world of HTTP, 'content' usually means stuff that's in the body of the response message. I recommend renaming 'content' to 'credentials' to align with the basic auth header terminology.